$OpenBSD: patch-CPUMeter_c,v 1.2 2018/07/26 12:49:04 ians Exp $

Don't allow getRange to set left bound equal to hw.ncpu, since it is used
as a loop condition. This eliminates a zero-index/off-by-one bug affecting
eight core machines specifically

Index: CPUMeter.c
--- CPUMeter.c.orig
+++ CPUMeter.c
@@ -135,6 +135,9 @@ static void AllCPUsMeter_getRange(Meter* this, int* st
          *count = cpus / 2;
          break;
    }
+
+   if (*count == cpus)
+      *count = *count - 1;
 }
 
 static void AllCPUsMeter_init(Meter* this) {
